packages/ui/src/forum/hooks/useCategoryLatestPost.ts

```diff
@@ -15,28 +17,29 @@ export const useCategoryLatestPost = (category_eq: string) => {
     fetchPolicy: 'cache-first',
   })
 
-  useEffect(() => {
-    const rawThread = threadData?.forumThreads[0]
-    if (rawThread)
-      fetchPost({
-        variables: {
-          where: { thread: { id_eq: rawThread.id } },
-          orderBy: ForumPostOrderByInput.UpdatedAtDesc,
-          limit: 1,
-        },
-      })
-  }, [threadData])
-
-  // const [fetchPost, { data: postData, loading: loadingPosts }] = useGetForumPostsLazyQuery()
-  const [fetchPost, { data: postData, loading: loadingPosts }] = Apollo.useLazyQuery(GetForumPostsDocument, {
-    fetchPolicy: 'cache-first',
-  })
+  // useEffect(() => {
+  //   const rawThread = threadData?.forumThreads[0]
+  //   if (rawThread)
+  //     fetchPost({
+  //       variables: {
+  //         where: { thread: { id_eq: rawThread.id } },
+  //         orderBy: ForumPostOrderByInput.UpdatedAtDesc,
+  //         limit: 1,
+  //       },
+  //     })
+  // }, [threadData])
 
-  const rawPost = postData?.forumPosts[0]
+  // const [fetchPost, { loading: loadingPosts }] = Apollo.useLazyQuery(GetForumPostsDocument, {
+  //   fetchPolicy: 'cache-first',
+  // })
 
-  return {
-    isLoading: loadingPosts || loadingThreads,
-    post: rawPost && asForumPost(rawPost),
-    thread: threadData?.forumThreads.length ? asForumThread(threadData?.forumThreads[0]) : undefined,
-  }
+  const [postsCache] = useLocalStorage<ForumPostsCache>()
+  return useMemo(
+    () => ({
+      isLoading: /*loadingPosts || */ loadingThreads,
+      post: postsCache?.posts.find((post) => post.categoryId === category_eq),
+      thread: threadData?.forumThreads.length ? asForumThread(threadData?.forumThreads[0]) : undefined,
```

Comment:
I think this is not needed:
```suggestion
```
just use this in `@/forum/components/category/LatestPost.tsx`:
```ts
      <LatestPostLink to={`${generatePath(ForumRoutes.thread, { id: post.threadId })}?post=${post.id}`}>
```